planning/costmap_generator/test/test_points_to_costmap.cpp

```diff
@@ -0,0 +1,226 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <costmap_generator/points_to_costmap.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <gtest/gtest.h>
+using pointcloud = pcl::PointCloud<pcl::PointXYZ>;
+class PointsToCostmapTest : public ::testing::Test
+{
+protected:
+  void SetUp() override { rclcpp::init(0, nullptr); }
+
+  ~PointsToCostmapTest() override { rclcpp::shutdown(); }
+
+  grid_map::GridMap constrcut_gridmap();
+
+public:
+  double grid_resolution_ = 1;
+  double grid_length_x_ = 20;
+  double grid_length_y_ = 20;
+  double grid_position_x_ = 0;
+  double grid_position_y_ = 0;
+};
+
+grid_map::GridMap PointsToCostmapTest::constrcut_gridmap()
+{
+  grid_map::GridMap gm;
+
+  // set gridmap size
+  gm.setFrameId("map");
+  gm.setGeometry(
+    grid_map::Length(grid_length_x_, grid_length_y_), grid_resolution_,
+    grid_map::Position(grid_position_x_, grid_position_y_));
+
+  // set initial value
+  gm.add("points", 0);
+
+  // set car postion in map frame to center of grid
+  grid_map::Position p;
+  p.x() = 0;
+  p.y() = 0;
+  gm.setPosition(p);
+
+  return gm;
+}
+
+// grid_y
+// |  map_x-------
+// |            |
+// |            |
+// |            |
+// |            map_y
+// |__________________grid_x
+TEST_F(PointsToCostmapTest, TestmakeCostmapFromPoints_validPoints)
+{
+  // construct pointcloud in map frame
+  pointcloud in_sensor_points;
+
+  in_sensor_points.width = 3;
+  in_sensor_points.height = 1;
+  in_sensor_points.is_dense = false;
+  in_sensor_points.resize(in_sensor_points.width * in_sensor_points.height);
+
+  in_sensor_points.points[0].x = 0.7;
+  in_sensor_points.points[0].y = 1;
+  in_sensor_points.points[0].z = 1;
+
+  in_sensor_points.points[1].x = 1.1;
+  in_sensor_points.points[1].y = 1;
+  in_sensor_points.points[1].z = 4;
+
+  in_sensor_points.points[2].x = 1.4;
+  in_sensor_points.points[2].y = 2;
+  in_sensor_points.points[2].z = 2.7;
+
+  grid_map::GridMap gridmap = constrcut_gridmap();
+
+  PointsToCostmap point2costmap;
+  double maximum_height_thres = 5;
+  double minimum_lidar_height_thres = 0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
+
+  std::string gridmap_layer_name = "points";
+  grid_map::Matrix costmap_data = point2costmap.makeCostmapFromPoints(
+    maximum_height_thres, minimum_lidar_height_thres, grid_min_value, grid_max_value, gridmap,
+    gridmap_layer_name, in_sensor_points);
+
+  int nonempty_gridcell_num = 0;
+  for (int i = 0; i < costmap_data.rows(); i++) {
+    for (int j = 0; j < costmap_data.cols(); j++) {
+      if (costmap_data(i, j) == grid_max_value) {
+        // std::cout << "i:"<< i <<",j:"<<j<< std::endl;
+        nonempty_gridcell_num += 1;
+      }
+    }
+  }
+
+  EXPECT_EQ(nonempty_gridcell_num, 3);
+}
+
+TEST_F(PointsToCostmapTest, TestmakeCostmapFromPoints_invalidPoints_biggerThanMaximumHeightThres)
+{
```

Comment:
Could you confirm the result of spelling checks in the diff view? :pray: 
https://github.com/autowarefoundation/autoware.universe/pull/960/files
![image](https://user-images.githubusercontent.com/31987104/170460917-c1f3e61e-bd49-4c88-b310-fc4025daaa5a.png)